packages/wdio-sauce-service/src/launcher.js

```diff
@@ -51,32 +51,36 @@ export default class SauceLauncher {
             }
         }
 
-        if (Array.isArray(capabilities)) {
-            for (const capability of capabilities) {
-                if (!capability['sauce:options']) {
-                    capability['sauce:options'] = {}
-                }
+        function prepareCapability(capability) {
+            const isW3C = !(capability.platform || capability.version) || capability['sauce:options']
+
+            if (!capability['sauce:options'] && isW3C) {
+                capability['sauce:options'] = {}
+            }
 
-                Object.assign(capability, endpointConfigurations)
+            Object.assign(capability, endpointConfigurations)
+            if (isW3C) {
```

Comment:
No need for that if we just declare a variable that defines the proper scope, e.g.:

```js
const capScope = isW3C ? capability['sauce:options'] : capabilities
capScope.tunnelIdentifier = ...
```

Reply:
👍

Reply:
Done